packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -30,26 +30,88 @@ export interface FluidAppInsightsLoggerConfig {
 		 */
 		mode: "inclusive" | "exclusive";
 		/**
-		 * Controls the default filtering of log events by their category.
+		 * Controls the filtering of log events.
 		 * Leaving this undefined will be treated as an empty array.
 		 */
 		filters?: TelemetryFilter[];
 	};
 }
 
 /**
- * Object used with an {@link FluidAppInsightsLoggerConfig}
- * to define logic for filtering of telemetry events
+ * A filter used to match against the category of a telemetry event
+ *
+ * @public
+ */
+export interface CategoryFilter {
+	/**
+	 * The categories of telemetry events that this filter applies to
+	 */
+	categories: TelemetryEventCategory[];
+}
+
+/**
+ * A filter used to match against the namespaces of a telemetry event
  *
  * @public
  */
-export interface TelemetryFilter {
+export interface NamespaceFilter {
+	/**
+	 * The namespace pattern to filter telemetry events.
+	 *
+	 * @remarks This will match namespaces that start with the given string. It is not a Regex pattern.
+	 * @example
+	 * "perf:latency" will match any namespace starting with "perf:latency"
+	 */
+	namespacePattern: string;
 	/**
-	 * The category of telemetry event that this filter applies to
+	 * A list of namespace patterns to explicitly exclude from the filter.
+	 *
+	 * @example If you have a namespacePattern of "perf:latency" but want to exclude
+	 *
+	 * events from "perf:latency:ops", you would add "perf:latency:ops" to this list.
 	 */
-	category: TelemetryEventCategory;
+	namespacePatternExceptions?: string[];
 }
 
+/**
```

Comment:
I'd rewrite the docs for this a bit. On one hand, the logic for picking which filter will apply for a given event; that's the namespace part, and here we also cover that the namespace exceptions allow for more selectivity. The filter with the most specific (longest) namespace pattern that matches an event is the one that gets used.

On the other hand, once the filter for an event is determined, what the filter does is evaluate the category in the event against the ones that are configured in it. And we can add a note that a filter with no namespace is just a catch-all for events that don't match any other filters.

Not for docs, but I imagine a common scenario would be to define specific filters for some namespaces where the user wants to see everything, but for anything not explicitly defined, they only want to see errors, so the last filter would be empty namespace and categories error.